bazel/scripts/setup_bazel_bash_completion.sh

```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env bash
+
+########################################################
+# Configures Bazel Bash-completion for the invoking user
+########################################################
+
+set -euo pipefail
+
+function log() {
+  echo "$@" >&2
+}
+
+function generate_completion() {
+  # outputs the completion script to stdout
+  #
+  # inspired by https://github.com/bazelbuild/bazel/blob/5f3f59ba367158b6c2e811f68cc946537b9b74d6/scripts/BUILD#L6-L26
+  # which is referenced in https://bazel.build/install/completion
+  curl https://raw.githubusercontent.com/bazelbuild/bazel/release-5.3.0/scripts/bazel-complete-header.bash
+  curl https://raw.githubusercontent.com/bazelbuild/bazel/release-5.3.0/scripts/bazel-complete-template.bash
+  bazel help completion
+}
+
+if ! [ -f ~/.bash_completion.d/bazel-complete.bash ]
+then
+  log "Creating completion script"
+  mkdir -p ~/.bash_completion.d
+  generate_completion > ~/.bash_completion.d/bazel-complete.bash
+else
+  log "Completion script already exists"
+fi
+
+if ! grep --quiet bazel-complete.bash ~/.bashrc
+then
+  log "Adapting ~/.bashrc to source completion script"
+  echo "source ~/.bash_completion.d/bazel-complete.bash" > ~/.bashrc
```

Comment:
was a little confused that there was no effect after executing the script manually on the devcontainer - should the bashrc (or only `bazel-complete.bash`) be sourced here explicitly?

Reply:
Sourcing wouldn't work, as this just affects the subshell that runs the script, not the shell that the script is invoked from. We could add a log telling the user to do that manually. That said, since usually this script isn't called directly by a user, but by an automatic provisioning system, I wouldn't bother.